### PR TITLE
bugfix/stdout-logs-utc

### PR DESCRIPTION
**Pull request by guybortnikov**
## Description
Currently, logs that are written to stdout are affected from the gProfiler instance environment timezone, this PR fixes it and changes the log records to be recorded in UTC time.

## Related Issue
Closes: https://github.com/Granulate/gprofiler/issues/595

## How Has This Been Tested?
```
docker run -it --env TZ=America/Los_Angeles --entrypoint /bin/bash --rm --userns=host --privileged --pid=host gprofiler
root@7eafc7ec4618:/app# date
Thu Nov 17 09:05:29 PST 2022
root@7eafc7ec4618:/app# python3 -m gprofiler -o /tmp -c
[17:05:42] Running gProfiler (version 1.11.0), commandline: '-o /tmp -c'
[17:05:42] gProfiler arguments: Namespace(alloc_interval=None, app_id_args_filters=[], application_metadata=True, collect_metadata=True, collect_metrics=True, config=None, container_names=True, continuous=True, controller_pid=None, curlify_requests=False, databricks_job_name_as_service_name=False, dotnet_mode='dotnet-trace', duration=60, flamegraph=True, frequency=11, identify_applications=True, insert_dso_name=False, java_async_profiler_args=None, java_async_profiler_mcache=30, java_async_profiler_mode='auto', java_async_profiler_safemode=64, java_collect_spark_app_name_as_appid=False, java_jattach_timeout=30, java_mode='ap', java_safemode='profiled-oom,profiled-signaled,hserr', java_version_check=True, log_file='/var/log/gprofiler/gprofiler.log', log_rotate_backup_count=1, log_rotate_max_size=5242880, log_to_server=True, log_usage=False, nodejs_mode='disabled', output_dir='/tmp', perf_dwarf_stack_size=8192, perf_inject=False, perf_mode='smart', perf_node_attach=False, php_mode='disabled', php_process_filter='php-fpm', pid_file='/var/run/gprofiler.pid', pid_ns_check=True, profile_api_version=None, profile_spawned_processes=False, profiling_mode='cpu', python_add_versions=True, python_mode='auto', python_pyperf_user_stacks_pages=None, rotating_output=False, ruby_mode='rbspy', server_host='https://profiler.granulate.io', server_token=None, server_upload_timeout=120, service_name=None, subcommand=None, upload_results=False, verbose=False)
[17:05:42] gProfiler Python version: 3.8.10 (default, Jun 22 2022, 20:18:18)
[GCC 9.4.0]
```

### CODE REVIEW

File: gprofiler/log.py [hunk 1]
```diff
@@ -199,6 +200,10 @@ def initial_root_logger_setup(
     logger_adapter = get_logger_adapter("gprofiler")
     logger_adapter.setLevel(logging.DEBUG)
 
+    # Patch formatTime to be GMT (UTC) for all formatters,
```
Comment (Jongy) [inline gprofiler/log.py:203]: If possible easily, I'd do it on the particular formatter object and not on the class. Such global will make it harder if we ever try to combine gProfiler's code into other projects.

**Review by Jongy — APPROVED** [review]
One small comment, please do if easy, otherwise can just merge.

I tested with an exe build on my IDT-TZ box, and now it's truly 2 hrs behind

File: gprofiler/log.py (hunk 1 above)
Comment (guybortnikov) [reply gprofiler/log.py]: Done

File: gprofiler/log.py [hunk 2]
```diff
@@ -189,6 +190,12 @@ def format(self, record: LogRecord) -> str:
         return formatted
 
 
+class GProfilerFormatter(ExtraFormatter):
+    # Patch formatTime to be GMT (UTC) for all formatters,
+    # see https://docs.python.org/3/library/logging.html?highlight=formattime#logging.Formatter.formatTime
+    converter = time.gmtime
```
Comment (Jongy) [inline gprofiler/log.py:196]: Hmm just add in `ExtraFormatter` no? It's already specific to gProfiler, why another `GProfilerFormatter`.

If you wanna write the "pretty thing" then I'd write a `UTCFormatter` class like this, and use it as a mixin in:
```
class GProfilerFormatter(UTCFormatter, ExtracFormatter): pass
```

Comment (Jongy) [reply gprofiler/log.py:196]: Also, pls rename `_ExtraFormatter` if it's not to be used directly anymore (only `GProfilerFormatter` is)

Comment (guybortnikov) [reply gprofiler/log.py:196]: Hmm, ok going with the mixin

**Review by Jongy — APPROVED** [review]
Tested locally again, LGTM

**Pull request merged**